fabfile.py

```diff
@@ -1,197 +1,518 @@
-                        )
-                else:
-                    local("git fetch origin")
-
-                with lcd(package_name):
-                    local("git checkout {}".format(version))
-                    local("git pull origin {}".format(version))
-                    local("python setup.py develop")
-
-
-def db_commands(username):
-    cmds =  [
-        "python manage.py migrate",
-        "python manage.py loaddata data/elcid.teams.json",
-        "python manage.py load_lookup_lists -f data/lookuplists/lookuplists.json",
-        "python manage.py create_random_data"
-    ]
-    python_cmd = "echo '"
-    python_cmd += "from django.contrib.auth.models import User; "
-    python_cmd += "User.objects.create_superuser(\"{0}\", \"{0}@example.com\", \"{0}1\")".format(username)
-    python_cmd += "' | python ./manage.py shell"
-    cmds.append(python_cmd)
-    return cmds
-
-def push_to_heroku(remote_name):
-    with lcd(fabfile_dir):
-        current_branch_name = local(
-            "git rev-parse --abbrev-ref HEAD", capture=True
+    #  https://stackoverflow.com/questions/14549270/check-if-database-exists-in-postgresql-using-shell
+    command = "sudo -u postgres psql -lqt | cut -d \| -f 1 | grep -w {database_name} | wc -l"
+    database_exists = "0" not in local(
+        command.format(database_name=some_env.database_name), capture=True
+    )
+
+    if database_exists:
+        postgres_command("DROP DATABASE {0}".format(some_env.database_name))
+
+    postgres_command("CREATE DATABASE {0}".format(some_env.database_name))
+    postgres_command("GRANT ALL PRIVILEGES ON DATABASE {0} TO {1}".format(
+        some_env.database_name, DB_USER
+    ))
+
+
+def postgres_load_database(backup_name, new_env):
+    if not os.path.isfile(backup_name):
+        raise ValueError("unable to find a backup at {}".format(
+            backup_name)
+        )
+
+    local("sudo -u postgres psql -d {0} -f {1}".format(
+        new_env.database_name,
+        backup_name
+    ))
+
+
+def services_symlink_nginx(new_env):
+    abs_address = "{}/etc/nginx.conf".format(new_env.project_directory, "")
+    if not os.path.isfile(abs_address):
+        raise ValueError(
+            "we expect an nginx conf to exist at {}".format(abs_address)
         )
-        local("git push {0} {1}:master".format(
-            remote_name, current_branch_name)
+
+    symlink_name = '/etc/nginx/sites-enabled/{}'.format(MODULE_NAME)
+    if os.path.islink(symlink_name):
+        local("sudo rm {}".format(symlink_name))
+
+    local('sudo ln -s {0} {1}'.format(abs_address, symlink_name))
+
+
+def services_symlink_upstart(new_env):
+    abs_address = "{}/etc/upstart.conf".format(new_env.project_directory, "")
+    if not os.path.isfile(abs_address):
+        raise ValueError(
+            "we expect an upstart conf to exist {}".format(abs_address)
         )
+    symlink_name = '/etc/init/{}.conf'.format(PROJECT_NAME)
+    local("sudo rm -f {}".format(symlink_name))
```

Comment:
Why do we do this rather than checking if it's a link the way we do for nginx.conf?

(I think I also want a log/print here if we remove something)